engines/python/setup/djl_python/transformers_neuronx.py

```diff
@@ -118,8 +119,14 @@ def set_model_loader_class(self) -> None:
             None
         """
         if self.config.model_loader == "tnx":
-            self._model_loader_class = TNXModelLoader
-            logging.info("Loading model using TNXModelLoader...")
+            if self.config.speculative_draft_model and self.config.rolling_batch == "vllm":
+                self._model_loader_class = TNXVllmModelLoader
+                logging.info(
+                    "Loading model using TNXVllmModelLoader for speculative decoding..."
+                )
+            else:
+                self._model_loader_class = TNXModelLoader
```

Comment:
how does this work if speculative model is set, but rolling batch is not vllm?

Reply:
Yes, it could work with TNX spec dec with batch size 1 though, I believe. We already have integration tests for this use case.